### PR TITLE
SONARJAVA 2219: Added addional Annotation to spring check

### PR DESCRIPTION
**Pull request by aepfli**
* added @Resource as injection Annotation
 * skipped @Inject as imho with @Inject there would be more necessary
   checks, like the qualifier in the class etc.

<details>
Please ensure your pull request adheres to the following guidelines: 

- [x ] Use the following formatting style: [SonarSource/sonar-developer-toolset](https://github.com/SonarSource/sonar-developer-toolset#code-style)
- [x ] Unit tests are passing and you provided a unit test for your fix
- [ ] ITs should pass : To run ITs locally, checkout the [README](https://github.com/SonarSource/sonar-java/blob/master/README.md) of the project. -> not sure if i need this
- [x ] If there is a [Jira](http://jira.sonarsource.com/browse/SONARJAVA) ticket available, please make your commits and pull request start with the ticket number (SONARJAVA-XXXX)

</details>

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/spring/SpringComponentWithNonAutowiredMembersCheck.java [hunk 1]
```diff
@@ -46,11 +46,16 @@ public void visitNode(Tree tree) {
...
         .map(m -> (VariableTree) m)
         .filter(v -> !v.symbol().isStatic())
-        .filter(v -> !v.symbol().metadata().isAnnotatedWith("org.springframework.beans.factory.annotation.Autowired"))
-        .forEach(v -> reportIssue(v.simpleName(), "Make this member @Autowired or remove it."));
+        .filter(v -> !isSpringInjectionAnnotated(v.symbol().metadata()))
+        .forEach(v -> reportIssue(v.simpleName(), "Make this member injected via Annotation (@Autowired, @Resource) or remove it."));
     }
   }
 
+  private boolean isSpringInjectionAnnotated(SymbolMetadata metadata) {
```
Comment (sonartech-externalpr) [inline java-checks/src/main/java/org/sonar/java/checks/spring/SpringComponentWithNonAutowiredMembersCheck.java]: MAJOR Make "isSpringInjectionAnnotated" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by sonartech-externalpr** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: java-checks/src/main/java/org/sonar/java/checks/spring/SpringComponentWithNonAutowiredMembersCheck.java (hunk 1 above)
Comment (aepfli) [inline java-checks/src/main/java/org/sonar/java/checks/spring/SpringComponentWithNonAutowiredMembersCheck.java]: i love that and i will do that :D

**Comment by m-g-sonar** [comment]
Hey @aepfli,

Thanks for your contribution. The specification for the rule has been updated to also cover `@Inject` annotation (see here: [RSPEC-3749](https://jira.sonarsource.com/browse/RSPEC-3749)), as discussed in the corresponding [thread from our ML](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/sonarqube/T-f83S9mvQU/xUZJIBceBAAJ).

I would appreciate if you also cover usage of the `@Inject` annotation, or at least give details about what makes it hard to cover.

**Comment by aepfli** [comment]
hi @m-g-sonar 

Well i was just not sure about @Inject but i read the docs and some blogs, and i am pretty sure now, that this is covered 

I changed the pull request and please review!

Thanks
Simon

[0] https://blogs.sourceallies.com/2011/08/spring-injection-with-resource-and-autowired/

**Comment by m-g-sonar** [comment]
Hello,

Technically, your code looks good to me. However, I just checked our description of the rule (See [RSPEC-3749](https://jira.sonarsource.com/browse/RSPEC-3749)) and it seems that the issue message which is expected is not the same as the one you are using.

RSPEC is saying : `"Annotate this member with "@Autowired", "@Resource", or "@Inject", or remove it."`
While you are using: `"Make this member injected via Annotation (@Autowired, @Inject, @Resource) or remove it."`

Once you updated the message, I'll open a duplicate of your PR in order to trigger our internal QA. If everything goes green, I'll then merge your PR.

Cheers,
Michael

**Comment by aepfli** [comment]
hi @m-g-sonar, 

i am really sorry, for causing so much "trouble". I did not think that the message is that critical, as the meaning is the same.

But i changed it, and i hope we are good to go.

I have to say, i am not sure if i have to escape the quotes in the "test-file" too, so for now i left it like that

**Comment by m-g-sonar** [comment]
No worries, no troubles at all. We try to specify as much as we can our rules, and the message is part of the equation.

Everything now LGTM. I'll open a PR duplicating your work to trigger our QA.

About your questions with the quotes, note that if you had escaped them in the test file, the test should have failed. The test files contains the issue messages "as they should look" when reported. :)

**Comment by m-g-sonar** [comment]
And... Everything went fine in our QA. Thanks again @aepfli for your contribution!
The fix will be delivered with SonarJava 4.9, expected for end of the week, or beginning of the next!

Cheers,
Michael

**Pull request merged**